steamtinkerlaunch

```diff
@@ -2384,6 +2385,110 @@ function fillProtonCSV {
 	fi
 }
 
+## Get internal name for a Proton version, first by checking for a 'compatibilitytool.vdf' file in its root directory, then for a Proton version file
+## Right now this is only used by addNonSteamGame
+## TODO at some point maybe we should store this in the ProtonCSV as well? Then the format would be 'versionfilename;protonpath;internalname'
+function getProtonInternalName {
+	## Tools are not necessarily guaranteed to have this comment, but I checked several and they all had it:
+	## - SteamTinkerLaunch (naturally)
+	## - All GE-Proton8 releases
+	## - Standard Proton-tkg releases
+	## - Luxtorpeda
+	##
+	## Steam Linux Runtime 1.0 (scout) / Native Linux Steam Linux Runtime is identified as 'steamlinuxruntime'
+	## No idea where the Steam Client gets this from, maybe it's just hardcoded, I couldn't find a string anywhere in the SteamLinuxRuntime installation folder or the 'appmanifest_1070560.acf'
+	function getProtonInternalNameVdf {
+		CTVPATH="$1"
+		if [ -f "$CTVPATH" ]; then
+			grep -i "// internal" "$CTVPATH" | sed 's-// Internal name of this tool--' | xargs
+		else
+			writelog "WARN" "${FUNCNAME[0]} - Could not find compatibilitytool.vdf file for Proton version at '$CTVPATH'"
+			echo ""
+		fi
+	}
+
+	## Get the Proton version version text file
+	function getProtonInternalNameVersionFile {
+		PPVPATH="$1"
+		if [ -f "$PPV" ]; then
+			awk '{print $2}' < "$PPV"
+		else
+			writelog "WARN" "${FUNCNAME[0]} - Could not find Proton version file at '$PPVPATH'"
+		fi
+	}
+
+	## Check if the path provided is for a Valve Proton version, by making some assumptions around the directory structure
+	function checkIsValveProton {
+		VPP="$1"  # Valve Proton Path
+
+		writelog "INFO" "${FUNCNAME[0]} - Checking if Proton version at '$VPP' is a Valve Proton version"
+		if [ -d "$VPP/dist" ] && [ ! -f "$VPP/$CTVDF" ]; then
+			writelog "INFO" "${FUNCNAME[0]} - Looks like we have a Valve Proton release here"
+			return 0
+		else
+			writelog "INFO" "${FUNCNAME[0]} - Doesn't look like a Valve Proton release, no 'dist' folder in the Proton path and we have a '$CTVDF'"
```

Comment:
This block will enter in all 3 cases not captured in the former
- VPP/dist exists & vpp/ctvdf exists
- VPP/dist does not exist & vpp/ctvdf exists
- VPP/dist does not exist & vpp/ctvdf does not exist

As long as thats the intended function all good, I just didn't get that from the log comment.

Reply:
Hmm, that doesn't sound like correct behaviour if I'm understanding.

`checkIsValveProton` should return `0` (success) if `VPP/dist` is a directory `&&` if `VPP/CTVDF` isn't a file, otherwise return `1` (non-success).

Maybe I'm making a silly mistake somewhere?

Reply:
Right, it will do that, but && means both of these things must be true at the same time to succeed.
The only case that I didn't mention was the success case which you laid out there.

true && true -> true
true && false -> false
false && true -> false
false && false -> false

If you want to change the logic on one being dependent on another you could do that in nested if blocks. Otherwise if you wanted to confirm what the comment is suggesting "No dist folder && yes CTVDF file", here is the full gambit... 

```
		if [ -d "$VPP/dist" ] && [ ! -f "$VPP/$CTVDF" ]; then
			writelog "INFO" "${FUNCNAME[0]} - Looks like we have a Valve Proton release here"
		elif [ ! -d "$VPP/dist" ] && [ -f "$VPP/$CTVDF" ]; then
			writelog "INFO" "${FUNCNAME[0]} - Doesn't look like a Valve Proton release, no 'dist' folder in the Proton path and we have a '$CTVDF'"
		elif [ -d "$VPP/dist" ] && [ -f "$VPP/$CTVDF" ]; then
                        echo "I don't know what having both of these implies"
		elif [ ! -d "$VPP/dist" ] && [ ! -f "$VPP/$CTVDF" ]; then
                        echo "I don't know what having neither of these implies                        
                fi
```

Reply:
Ahhh I see what you mean now. The log comment isn't very accurate then, since it would still trigger if `! -d "$VPP/dist"`. Didn't even catch that myself! I will probably just change the log statement slightly, because the logic is right but the comment isn't.

Really, an `||` might work here, but to be extra sure I'm using `&&` because it's a better way to ensure a Valve Proton (except for Experimental/Hotfix, which uses `files` like community Proton flavours instead of `dist`, but we're hardcoding that earlier anyway). 

Thanks for the review on this and keeping things on the right track, very much appreciated :-)